pkg/security/probe/probe.go

```diff
@@ -329,8 +343,66 @@ func (p *Probe) DispatchEvent(event *model.Event) {
 		handler.HandleEvent(event)
 	}
 
-	// Process after evaluation because some monitors need the DentryResolver to have been called first.
-	p.monitor.ProcessEvent(event)
+	// TODO: split this func
+	if p.Config.SecurityProfileAnomalyDetection && event.ProfileState == model.MatchedAndAbsent && event.GetEventType() != model.SyscallsEventType {
+		p.monitor.securityProfileManager.FillProfileContextFromContainerID(event.ProcessContext.ContainerID, &event.SecurityProfileContext)
+		if event.GetEventType() == model.FileOpenEventType {
+			if p.Config.SecurityProfileFilesBestEffort {
+				fmt.Printf("FILE Event not found in profile -> generate anomaly detection ! %s @ %+v for %s\n",
+					event.ProcessContext.Comm, event.GetEventType().String(), event.Open.File.PathnameStr)
+				if p.anomalyDetectionRateLimiterOpen.Allow() {
+					p.DispatchCustomEvent(
+						events.NewCustomRule(events.AnomalyDetectionRuleID),
+						events.NewCustomEvent(model.AnomalyDetectionOpenEventType, serializers.NewEventSerializer(event, p.resolvers)),
```

Comment:
See my comment below, I think we can simplify this function by removing the `AnomalyDetectionOpenEventType` event types.